velox/exec/TableWriter.h

```diff
@@ -113,24 +113,17 @@ class TableWriter : public Operator {
 
   void addInput(RowVectorPtr input) override;
 
-  void noMoreInput() override {
-    Operator::noMoreInput();
-    if (aggregation_ != nullptr) {
-      aggregation_->noMoreInput();
-    }
-    close();
-  }
+  void noMoreInput() override;
 
   virtual bool needsInput() const override {
     return true;
   }
 
   void close() override {
     if (!closed_) {
-      if (dataSink_ != nullptr) {
-        dataSink_->close();
-      }
-      closed_ = true;
+      // Abort the data sink if the query has already failed and no need for
```

Comment:
Do we need to call close on aggregation_ ?

Reply:
Good catch. Yeah, we should!